etl/helpers.py

```diff
@@ -118,3 +151,110 @@ def walden_dataset(self) -> WaldenDataset:
+        """
+        if namespace is None:
+            # If namespace is not specified, catch any name that does not contain "/".
+            namespace = r"[^/]+"
+
+        if version is None:
+            # If version is not specified, catch any version, which could be either a date, a year, or "latest".
+            version = r"(?:\d{4}\-\d{2}\-\d{2}|\d{4}|latest)"
+
+        if channel in ["meadow", "garden", "grapher"]:
+            step_name = f"data://{channel}/{namespace}/{version}/{short_name}"
+        elif channel in ["snapshot", "walden"]:
+            step_name = f"{channel}://{namespace}/{version}/{short_name}"
+        elif channel is None:
+            step_name = rf"(?:snapshot:/|walden:/|data://meadow|data://garden|data://grapher)/{namespace}/{version}/{short_name}"
+        else:
+            raise UnknownChannel
+
+        return step_name
+
+    @staticmethod
+    def _get_attributes_from_step_name(step_name: str) -> Dict[str, str]:
+        """Get attributes (channel, namespace, version and short name) from the step name (as it appears in the dag)."""
+        channel_type, path = step_name.split("://")
+        if channel_type in ["walden", "snapshot"]:
+            channel = channel_type
+            namespace, version, short_name = path.split("/")
+        elif channel_type in ["data"]:
+            channel, namespace, version, short_name = path.split("/")
+        else:
+            raise WrongStepName
+
+        attributes = {"channel": channel, "namespace": namespace, "version": version, "short_name": short_name}
+
+        return attributes
+
+    @property
+    def step(self) -> str:
+        return self._create_step_name(
+            channel=self.channel, namespace=self.namespace, version=self.version, short_name=self.short_name
+        )
+
+    @property
+    def dependencies(self) -> List[str]:
+        return DAG[self.step]
+
+    def get_dependency_step_name(
+        self,
+        short_name: str,
+        channel: Optional[str] = None,
+        namespace: Optional[str] = None,
+        version: Optional[Union[str, int]] = None,
+    ) -> str:
+        """Get dependency step name (as it appears in the dag) given its attributes (at least its short name)."""
+        pattern = self._create_step_name(channel=channel, namespace=namespace, version=version, short_name=short_name)
+        matches = [dependency for dependency in self.dependencies if bool(re.match(pattern, dependency))]
+
+        if len(matches) == 0:
+            raise NoMatchingStepsAmongDependencies
+        elif len(matches) > 1:
+            raise MultipleMatchingStepsAmongDependencies
+
+        dependency = matches[0]
+
+        return dependency
+
+    def load_dependency(
+        self,
+        short_name: str,
+        channel: Optional[str] = None,
+        namespace: Optional[str] = None,
+        version: Optional[Union[str, int]] = None,
+    ) -> Union[catalog.Dataset, Snapshot, WaldenCatalog]:
```

Comment:
I'd drop support for Walden and its catalog. Only snapshots need to be supported.

Reply:
Let's keep it while it still exists. Once we get rid of Walden I'll remove all mentions here.